webapp/maps_heatmaps/src/index.tsx

```diff
@@ -186,60 +190,67 @@ async function updateImagesAndMarkers(first: boolean): Promise<void> {
     allDocArrays[i] = await getNextDocs(i, first);
     pointers[i] = 0;
   }
-  if (elementById != null) {
-    elementById.innerHTML = "";
-    try {
-      while (countOfImagesAndMarkers < NUM_OF_IMAGES_AND_MARKERS) {
-        let minDocData, docData;
-        do {
-          docData = await getMinDoc(allDocArrays, pointers);
-          minDocData = docData.data();
-        } while (!isInVisibleMap(minDocData, map));
-        const latlng = convertGeopointToLatLon(minDocData.coordinates);
-        const imageElement = addImageToSidePanel(minDocData, elementById);
-        await addMarkerWithListener(
-          imageElement,
-          map,
-          latlng,
-          docData.id,
-          minDocData.labels,
-          selectedDate
-        );
-        countOfImagesAndMarkers++;
+  const nextBtn = document.getElementsByTagName("button").namedItem("next-btn");
+  if (nextBtn) nextBtn.disabled = false;
+  if (!timeOfRequest) {
+    timeOfRequest = Date.now();
+    timeOfLastRequest = timeOfRequest;
+  }
+  if (timeOfRequest === timeOfLastRequest) {
```

Comment:
Add a comment why we have this check (we ignore the results otherwise)

Reply:
Added,